go/vt/schemadiff/table_test.go

```diff
@@ -390,6 +390,13 @@ func TestCreateTableDiff(t *testing.T) {
 			from: "create table t1 (`id` int primary key, i int)",
 			to:   "create table t2 (`id` int, i int, primary key (id))",
 		},
+		{
+			name:  "change primary key column",
+			from:  "create table t1 (id int primary key, info int not null);",
+			to:    "create table t1 (the_id int primary key, info int not null);",
+			diff:  "alter table t1 drop primary key, drop column id, add column the_id int first, add primary key (the_id)",
+			cdiff: "ALTER TABLE `t1` DROP PRIMARY KEY, DROP COLUMN `id`, ADD COLUMN `the_id` int FIRST, ADD PRIMARY KEY (`the_id`)",
```

Comment:
I would have expected to rename the column. I guess not? I imagine you have valid reasons.

Reply:
Clarified in code why this is actually the expected behavior (there's a _hint_ switch to control what behavior you want)